SNIPS/snip-86.md

```diff
@@ -0,0 +1,601 @@
+    "revision": "1"
+  },
+  "message": {
+    "version": "1.0.0",
+    "address": "0x0987654321098765432109876543210987654321098765432109876543210987",
+    "statement": "Sign in with Starknet to access Test Application",
+    "uri": "https://test.example.org",
+    "nonce": "abcdef123456",
+    "issuedAt": "2024-05-24T14:00:00Z",
+    "notBefore": "2024-05-24T14:05:00Z",
+    "expirationTime": "2024-05-24T15:00:00Z"
+  },
+  "primaryType": "Message",
+  "types": {
+    "Message": [
+      { "name": "version", "type": "string" },
+      { "name": "address", "type": "felt" },
+      { "name": "statement", "type": "string" },
+      { "name": "uri", "type": "string" },
+      { "name": "nonce", "type": "string" },
+      { "name": "issuedAt", "type": "string" },
+      { "name": "notBefore", "type": "string" },
+      { "name": "expirationTime", "type": "string" }
+    ],
+    "StarknetDomain": [
+      { "name": "name", "type": "string" },
+      { "name": "chainId", "type": "string" },
+      { "name": "version", "type": "string" },
+      { "name": "revision", "type": "string" }
+    ]
+  }
+}
+```
+
+The following is an example of a SIWS message with all minimal fields:
+
+```json
+{
+  "domain": {
+    "chainId": "SN_MAIN",
+    "name": "Minimal App",
+    "version": "1.0.0",
+    "revision": "1"
+  },
+  "message": {
+    "version": "1.0.0",
+    "address": "0xabcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890",
+    "statement": "Sign in to Minimal App",
+    "uri": "https://minimal.app",
+    "nonce": "minimalnonce123",
+    "issuedAt": "2024-05-24T16:00:00Z"
+  },
+  "primaryType": "Message",
+  "types": {
+    "Message": [
+      { "name": "version", "type": "string" },
+      { "name": "address", "type": "felt" },
+      { "name": "statement", "type": "string" },
+      { "name": "uri", "type": "string" },
+      { "name": "nonce", "type": "string" },
+      { "name": "issuedAt", "type": "string" }
+    ],
+    "StarknetDomain": [
+      { "name": "name", "type": "string" },
+      { "name": "chainId", "type": "string" },
+      { "name": "version", "type": "string" },
+      { "name": "revision", "type": "string" }
+    ]
+  }
+}
+```
+
+### Signing and Verifiying Messages with Starknet Accounts
+
+#### Signing Messages with Starknet Accounts
+
+1. **Message Preparation**:
+
+   - **Structured Data**: Messages are structured in JSON format for clarity and integrity.
+   - **Poseidon Hash Function**: Used to handle larger data sizes, enabling the signing of messages of any length.
+
+2. **Generating the Signature**:
+
+   - **Account Abstraction**: Signature logic is defined by the account’s smart contract.
+   - **Signing Process**: User’s private key signs the hashed message securely within Starknet.
+
+#### Verifying Messages with Starknet Accounts
+
+1. **Verification Method for EOAs**:
+
+   - **EIP-712**: Verification involves reconstructing the signed message and comparing it against the provided signature.
```

Comment:
This is referencing EIP which aren't compatible with Starknet. Wouldn't it be better to reference the SNIPs?
I believe this is SNIP-12 and EIP-4361 is SNIP-6

Reply:
Agreed, included in 86b90d8